### PR TITLE
coll: refactoring nonblocking collectives

### PR DESCRIPTION
**Pull request by yfguo**
## Pull Request Description

This PR refactors all the nonblocking collectives following the same implementation as Ibcast. They use the new unified schedule interface and calls the sched functions for the corresponding algorithms.

The Gentran wrapper is removed as they are no longer in use.

This is a prereq for persistent collectives.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [ ] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by yfguo** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

File: src/mpi/rma/rmatypeutil.c [hunk 1]
```diff
@@ -48,8 +48,10 @@ int MPIR_Compare_equal(const void *a, const void *b, MPI_Datatype type)
 #define MPIR_OP_TYPE_MACRO(mpi_type_, c_type_, type_name_) \
         case mpi_type_:                         \
             if (*(c_type_ *)a == *(c_type_ *)b) \
+            {                                   \
```
Comment (hzhou) [inline src/mpi/rma/rmatypeutil.c:51]: Concatenate the brace

File: src/mpi/rma/rmatypeutil.c [hunk 2]
```diff
@@ -48,8 +48,10 @@ int MPIR_Compare_equal(const void *a, const void *b, MPI_Datatype type)
...
                 return TRUE;                    \
-                break;
+            }                                   \
+            break;
             MPIR_OP_TYPE_GROUP(C_INTEGER)
```
Comment (hzhou) [inline src/mpi/rma/rmatypeutil.c:55]: Probably an empty line to separate the macro definition from code is a good idea here.

Comment (yfguo) [reply src/mpi/rma/rmatypeutil.c:55]: Good point. Let me try if that works.

**Comment by hzhou** [comment]
test:mpich/warnings

**Comment by yfguo** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

**Comment by yfguo** [comment]
@hzhou This one is good to go. The tests were successful. I force-pushed the fix to the warning squash commit. The whitespace failure is false positive.

**Comment by hzhou** [comment]
> 
> 
> @hzhou This one is good to go. The tests were successful. I force-pushed the fix to the warning squash commit. The whitespace failure is false positive.

You can add 
```
/* *INDENT-OFF* */
...
/* *INDENT-ON* */
```
to exclude certain block of code from whitespace checker.

**Comment by yfguo** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

retesting to make sure I did not break anything during rebasing.

**Comment by yfguo** [comment]
test:mpich/ch4/ofi
test:mpich/ch3/ofi

**Review by hzhou — APPROVED** [review]
LGTM

**Pull request merged**